resources/aws/gateway.go

```diff
@@ -103,15 +88,18 @@ func (g *Gateway) CreateOrFail() error {
 }
 
 func (g *Gateway) Delete() error {
-	gateway, err := g.findExisting()
+	gateways, err := g.list()
 	if err != nil {
 		return microerror.MaskAny(err)
 	}
 
-	if _, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
-		InternetGatewayId: gateway.InternetGatewayId,
-	}); err != nil {
-		return microerror.MaskAny(err)
+	for _, gateway := range gateways {
```

Comment:
There will be always one gateway, assuming that someone didn't create manually some gateway with the same name - and I'm also 99% sure that creating manually such a conflicting gateway is just impossible. I think that it's a matter of whether we prefer to
- explicitly check if list is empty
- iterate over it
Which is a stylistical dilemma which I have no strong feelings about, so feel free to argue (and of course reach some consensus) if you do.

But after all, the list should always contain 0 or 1 element.